src/libcore/cell.rs

```diff
@@ -10,6 +10,23 @@
 
 //! Shareable mutable containers.
 //!
+//! Rust memory safety is based on this rule: Given an object `T`, it is only possible to
+//! have one of the following:
+//!
+//! - Having several immutable references (`&T`) to the object (also know as Aliasing).
+//! - Having one mutable reference (`&mut T`) to the object (also know as Mutability).
+//!
+//! This is enforced by the Rust compiler. However, there are situations where this rule is not
+//! flexible enough. Sometimes it is required to have multiple references to an object and yet
+//! mutate it.
+//!
+//! Shareable mutable containers exist to permit mutability in the presence of aliasing in a
```

Comment:
"in ___ in ___" reads a bit awkward. I might write:

**Before:** ...to permit mutability in the presence of aliasing in a controlled manner
**After:** ...to permit mutability in a controlled manner, even in the presence of aliasing.